csrc/device_lower/pass/insert_syncs.cpp

```diff
@@ -486,19 +486,13 @@ class ReadAfterWriteSyncs : public kir::ExprMutator {
         // async mma pipeline has not been flushed yet.
         flush_async_mma_pipeline_ = false;
       } else if (mma->isBlackwell()) {
-        // TODO: This is clearly a wrong way to sync, but as an intermediate
-        // step to enable incremental development, we use nanosleep to sync the
-        // mma. We should replace this with a correct sync method.
-        registerInsertBefore(expr, IrBuilder::create<kir::BlockSync>());
         registerInsertAfter(
             expr,
-            IrBuilder::create<kir::Asm>(
-                "nanosleep.u32",
-                std::vector<Val*>{},
-                std::vector<Val*>{
-                    IrBuilder::create<Val>(4000000000, DataType::UInt32)},
-                kir::Asm::Options{/*volatile=*/true}));
-        registerInsertAfter(expr, IrBuilder::create<kir::BlockSync>());
+            IrBuilder::create<kir::MBarrierWaitParity>(
```

Comment:
For TMA Load non-circular buffer, we add the mbarrier wait in the indexing pass.
https://github.com/NVIDIA/Fuser/blob/main/csrc/device_lower/pass/index.cpp#L1549

Is there a reason for adding mbarrier wait for `MmaOp` here? Just trying to maintain consistency.

Reply:
The difference is:

Doing in indexing:

```C++
if (elect-sync) {
  mma;
  commit;
  wait;
}
```

vs 

Doing here:

```C++
if (elect-sync) {
  mma;
  commit;
}
wait;
```

I just tried moving it to indexing, and see some misaligned address error. I will leave it as is here for now. I may come back to this topic the after I return from my PTO the week after next week when I started to work on schedulers.